README.md

```diff
@@ -1,486 +1,679 @@
+- Class expressions may be decorated, not just class declarations`.
+- Class decorators come after `export` and `default`.
 
-### `decorator @xyz` declarations
+There is no special syntax for defining decorators; any function can be applied as a decorator.
 
-Decorators may be defined as a simple composition of other decorators. You can use all the fancy JavaScript features you want inside the arguments, but at the top level, this is just a string of decorators and arguments for these decorators. There's no way to conditionally use one decorator in one situation and another in another situation, for example.
+# Detailed semantics 
 
-Example:
+The three steps of decorator evaluation:
+1. Decorator expressions and 
+s (the thing after the `@`) are *evaluated* interspersed with computed property names.
+1. Decorators are *called* (as functions) during class definition, after the methods have been evaluated but before the constructor and prototype have been put together.
+1. Decorators are *applied* (mutating the constructor and prototype) all at once, after all of them have been called.
+
+The semantics here generally follow the consensus at the May 2016 TC39 meeting in Munich.
+
+## 1. Evaluating decorators
+
+Decorators are evaluated as expressions, interspersed in their evaluation order with computed property names. This goes left to right, top to bottom. The result of decorators is stored in the equivalent of local variables to be later called after the class definition initially finishes executing.
+
+## 2. Calling decorators
+
+### The element being wrapped: the first parameter
+
+The first parameter, of what the decorator is wrapping, depends on what is being decorated:
+- In a method, init-method, getter or setter decorator: the relevant function object
+- In a class decorator: the class
+- In a field: An object with two properties
+    - `get`: A function which takes no arguments, expected to be called with a receiver which is the appropriate object, returning the underlying value.
+    - `set`: A function which takes a single argument (the new value), expected to be called with a receiver which is the object being set, expected to return `undefined`.
+
+### The context object: the second parameter
+
+The context object--the object passed as the second argument to the decorator--contains the following properties:
+- `kind`: One of
+    - `"class"`
+    - `"method"`
+    - `"init-method"`
+    - `"getter"`
+    - `"setter"`
+    - `"field"``
+- `name`:
+    - Public field or method: the `name` is the String or Symbol property key.
+    - Private field or method: missing (could be provided as some representation of the private name, in a follow-on proposal) 
+    - Class: missing
+- `isStatic`:
+    - Static field or method: `true`
+    - Instance field or method: `false`
+    - Class: missing
```

Comment:
why a boolean over a "placement" enum?

Reply:
sure, we could switch to that. I'd accept a PR to fix up this patch, or we can iterate on it later.